src/golang/cmd/server/handler/get_workflow.go

```diff
@@ -187,10 +191,28 @@ func (h *GetWorkflowHandler) Perform(ctx context.Context, interfaceArgs interfac
 
 	workflowDagResults := make([]workflowDagResult, 0, len(dagResults))
 	for _, dagResult := range dagResults {
+		var dagExecState shared.ExecutionState
+		if !dagResult.ExecState.IsNull {
```

Comment:
What is the scenario in which the DAG result would have a different status vs. execution state? That seems very strange.

Reply:
It also doesn't seem like we're doing anything other than logging that information — is that intentional?

Reply:
yeah it's mostly just a passive check. It definitely shouldn't happen but we don't technically validate this anywhere. Once https://linear.app/aqueducthq/issue/ENG-2665/remove-redundant-status-field-in-workflowdagresult-responses is addressed we won't need to think about this at all.

Reply:
Got it. Should we tag this to be removed once ENG-2665 is addressed?

Reply:
yes, can do